x-pack/plugins/security_solution/public/common/components/visualization_actions/use_save_to_library.test.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { renderHook, act } from '@testing-library/react-hooks';
+import { toMountPoint } from '@kbn/react-kibana-mount';
+import { useSaveToLibrary } from './use_save_to_library';
+import { useKibana } from '../../lib/kibana';
+import { useRedirectToDashboardFromLens } from './use_redirect_to_dashboard_from_lens';
+import { useGetSecuritySolutionUrl } from '../link_to';
+import { kpiHostMetricLensAttributes } from './lens_attributes/hosts/kpi_host_metric';
+
+jest.mock('../../lib/kibana', () => ({
+  useKibana: jest.fn(),
+}));
+
+jest.mock('./use_redirect_to_dashboard_from_lens', () => ({
+  useRedirectToDashboardFromLens: jest.fn(),
+}));
+
+jest.mock('../link_to', () => ({
+  useRedirectToDashboardFromLens: jest.fn(),
+  useGetSecuritySolutionUrl: jest.fn(),
+}));
+
+jest.mock('@kbn/react-kibana-mount', () => ({
+  toMountPoint: jest.fn(),
+}));
+
+describe('useSaveToLibrary hook', () => {
+  const mockStartServices = {
+    application: { capabilities: { visualize: { save: true } } },
+    lens: { SaveModalComponent: jest.fn() },
+  };
+
+  const mockGetSecuritySolutionUrl = jest.fn();
+  const mockRedirectTo = jest.fn();
+  const mockGetEditOrCreateDashboardPath = jest.fn().mockReturnValue('mockDashboardPath');
+
+  beforeEach(() => {
+    (useKibana as jest.Mock).mockReturnValue({ services: mockStartServices });
+    (useRedirectToDashboardFromLens as jest.Mock).mockReturnValue({
+      redirectTo: mockRedirectTo,
+      getEditOrCreateDashboardPath: mockGetEditOrCreateDashboardPath,
+    });
+    (useGetSecuritySolutionUrl as jest.Mock).mockReturnValue(mockGetSecuritySolutionUrl);
+    (toMountPoint as jest.Mock).mockImplementation(() => jest.fn());
+  });
```

Comment:
Can we clean up this a little bit:
- If the mocks are not overridden inside the tests, there's no need to redefine them every time inside the `beforeEach`. They can be moved outside.
- If the mock is never used, there's no need to create a const for it, like `mockGetSecuritySolutionUrl`, `mockRedirectTo`, and `mockGetEditOrCreateDashboardPath`. We can just do `(useGetSecuritySolutionUrl as jest.Mock).mockReturnValue(jest.fn());` only once.
- If mocks are used, it would be better to create a const for it, like `useKibana`, we can create the `const mockUseKibana = useKibana as jest.Mock`, so we don't have to cast it every time we use it.